modules/search/client/components/SearchUsers.component.js

```diff
@@ -0,0 +1,182 @@
+
+import UsersList from './UsersList';
+
+// import { searchUsers } from './search.api';
+import { searchUsers } from '@/modules/users/client/api/search-users.api.js';
+
+function UsersResults({ users, resolved }) {
+  const userList = (
+    <div className="contacts-list" ng-if="ContactsList.contacts.$resolved && ContactsList.contacts.length > 0">
+
+      <div className="row">
+        <div className="col-xs-12" ng-class="{'col-sm-8': ContactsList.contacts.length >= 6, 'text-center': ContactsList.contacts.length < 6}">
+          <h4 className="text-muted">
+
+            {users.length === 1 &&
+            <span>
+            One user found.
+            </span>
+            }
+
+            {users.length > 1 &&
+            <span>
+              {users.length} users found.
+              {/* {{ (ContactsList.contacts | filter:{ confirmed: true }).length }} contacts */}
+            </span>
+            }
+
+            <UsersList users={users}/>
+
+          </h4>
+        </div>
+      </div>
+    </div>
+  );
+
+  const noUsers = (
+    <div className="row content-empty" ng-if="ContactsList.contacts.$resolved && !ContactsList.contacts.length">
+      <i className="icon-3x icon-users"></i>
+      <h4>No users found with this name.</h4>
+    </div>
+  );
+
+  return (
+    <div>
+      {!resolved?
+        <div className="content-wait"
+          role="alertdialog"
+          aria-busy="true"
+          aria-live="assertive"
+          ng-if="!ContactsList.contacts || !ContactsList.contacts.$resolved">
+          <small>Wait a moment...</small>
+        </div>
+        :
+        (users && users.length > 0? userList : noUsers)
+      }
+    </div>
+  );
+}
+
+UsersResults.propTypes = {
+  users: PropTypes.array,
+  resolved: PropTypes.bool
+};
+
+// export class SearchUsersBar extends React.Component {
+
+// export default function SearchUsers({ searchQuery }) {
+class SearchUsers extends React.Component {
+
+  constructor(props) {
+    super(props);
+    console.log('SearchUsers constructor');
+    this.handleChange = this.handleChange.bind(this);
+    this.actionSearch = this.actionSearch.bind(this);
+
+    this.state = {
+      searchQuery: props.searchQuery || '',
+      resolved: true
+    };
+  }
+
+  handleChange(event) {
+    this.setState({ searchQuery: event.target.value });
+    // this.props.searchQuery = event.target.value ;
+  }
+
+  async actionSearch(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.setState({ resolved: false });
+    console.log('Searching for: ' + this.state.searchQuery);
+
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    const userResults = await searchUsers(this.state.searchQuery);
+    console.log(userResults);
+    this.setState({ resolved: true, users: userResults.data });
+    console.log(this.state.users);
+  };
+
+  clickClear() {
+    this.setState({ searchQuery: '' });
+  }
+
+  render() {
+    console.log('SearchUsers: render');
```

Comment:
Let's remember to remove `console.log`s from the PR before merging.

Reply:
sure :+1:

Reply:
[eslint's `no-console`](https://eslint.org/docs/rules/no-console) can be interesting for us.

Just disable it explicitly whenever we really need it with `eslint-disable-line`.

Reply:
Valid point @mrkvon ! 👍 :shipit:

Reply:
@mrkvon added the rule in https://github.com/Trustroots/trustroots/pull/1015